app/src/main/java/de/westnordost/streetcomplete/util/logs/DatabaseLogger.kt

```diff
@@ -0,0 +1,56 @@
+package de.westnordost.streetcomplete.util.logs
+
+import de.westnordost.streetcomplete.data.logs.LogLevel.*
+import de.westnordost.streetcomplete.data.logs.LogMessage
+import de.westnordost.streetcomplete.data.logs.LogsDao
+import de.westnordost.streetcomplete.util.ktx.nowAsEpochMilliseconds
+import kotlinx.coroutines.CoroutineName
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.launch
+
+class DatabaseLogger(private val logsDao: LogsDao) : Logger {
```

Comment:
It is quite consistently implemented in the data layer within StreetComplete that the only place a DAO is accessed from the outside is its associated `Controller` (if there is one). The reason is that if (in the future) there shall be a listener that observes updates on the log such as when a new log message is added - this listener would be on the `LogsController`. To be able to fire the even to its listeners, the `LogsController` must know about it whenever the log changes.*

So, for consistency, please use `LogsController` here as a dependency and implement the adding of a `LogMessage` also in `LogsController`.

* actually, for the UI it would make sense if there was such a listener - to have live updates of the log in the log view, no?